src/background.js

```diff
@@ -25,6 +25,49 @@ async function initApi() {
   return api !== null;
 }
 
+/* Dynamic badge */
+function setStarredBadge(tabId) {
+  browser.browserAction.setBadgeText({ text: "★", tabId: tabId });
+  browser.browserAction.setBadgeTextColor({ color: "#FFE234", tabId: tabId });
+  browser.browserAction.setBadgeBackgroundColor({ color: "rgba(102,6,0,0.3)", tabId: tabId });
+  browser.browserAction.setTitle({ title: "Edit bookmark", tabId: tabId });
+}
+
+function resetStarredBadge(tabId) {
+  browser.browserAction.setBadgeText({ text: "", tabId: tabId });
+  browser.browserAction.setTitle({ title: "Save bookmark", tabId: tabId });
+}
+
+async function setDymamicBadge(tabId) {
+  const badgeText = await browser.browserAction.getBadgeText({ tabId });
+  const starred = badgeText === "★";
+
+  const tab = await browser.tabs.get(tabId);
+  const tabMetadata = await loadTabMetadata(tab.url);
```

Comment:
This will fetch the metadata for every single navigated URL from the linkding server, which was disabled by default as part of a previous issue and PR:
- https://github.com/sissbruecker/linkding-extension/issues/36
- https://github.com/sissbruecker/linkding-extension/pull/38

This new feature should only be active if the respective option for pre-caching requests is enabled (`precacheEnabled`). The option can be respected by passing `true` as the second parameter: `loadTabMetadata(tab.url, true)`.

The options page should also be modified somehow so that it becomes apparent that the star badge will only work if the option is enabled.

Reply:
There are also existing listeners for tab activated / updated, which already prefetch the metadata. It would be good if the extension wouldn't fetch the metadata twice when a tab is activated / updated.